component/loki/process/stages/drop.go

```diff
@@ -116,72 +137,90 @@ func (m *dropStage) shouldDrop(e Entry) bool {
 		ct := time.Now()
 		if e.Timestamp.Before(ct.Add(-m.cfg.OlderThan)) {
 			// Too old, drop
-			level.Debug(m.logger).Log("msg", fmt.Sprintf("line met drop criteria for age; current time=%v, drop before=%v, log timestamp=%v", ct, ct.Add(-m.cfg.OlderThan), e.Timestamp))
+			if Debug {
+				level.Debug(m.logger).Log("msg", fmt.Sprintf("line met drop criteria for age; current time=%v, drop before=%v, log timestamp=%v", ct, ct.Add(-m.cfg.OlderThan), e.Timestamp))
+			}
 		} else {
-			level.Debug(m.logger).Log("msg", fmt.Sprintf("line will not be dropped, it did not meet drop criteria for age; current time=%v, drop before=%v, log timestamp=%v", ct, ct.Add(-m.cfg.OlderThan), e.Timestamp))
+			if Debug {
```

Comment:
Why does the debug flag exist? Is this to reduce log file creation?

Reply:
Yes. Debug logging in Promtail code can be a HUGE CPU hog (like 3 CPU cores maxed out).

Also, it was only ever used by tests. This is a temporary failsafe to avoid actually suffocating any environments that happen to run debug-level logs and use some `loki.*` component.

It's in our plans to improve the overall logging behavior and overhead of the logging components.

Reply:
Gotcha, not in love with it but it is what it is.